python/ray/tune/function_runner.py

```diff
@@ -350,10 +350,18 @@ def _report_thread_runner_error(self, block=False):
             pass
 
 
-def detect_checkpoint_function(train_func):
+def detect_checkpoint_function(train_func, abort=False):
+    """Use checkpointing if any arg has "checkpoint" and args = 2"""
     func_args = inspect.getfullargspec(train_func).args
-    use_checkpoint = "checkpoint" in func_args
-    return use_checkpoint
+    validated = len(func_args) == 2 and any("checkpoint" in arg
+                                            for arg in func_args)
+    if abort and not validated:
```

Comment:
```suggestion
    validated = validated or (len(func_args) == 1) and any(
        "checkpoint" in arg for arg in func_kwargs)
    if abort and not validated:
```

Reply:
With this addition we enable `functools.partial` to work. E.g.

```
def train(config, data_dir=None, checkpoint=None):
    pass
detect_checkpoint_function(partial(train, data_dir="/tmp"))  # True
```

I came about this problem a few times in some of the examples. Since we pass `checkpoint` explicitly we should gain some flexibility with this